### ISSUE TITLE
NullPointerException with GrizzlyAsyncHttpProvider(v1.8) when WebSocket is closed

### ISSUE THREAD

**Issue by laszlohordos** [AsyncHttpClient/async-http-client#829]
The Grizzly ProtocolHandler (grizzly-websockets-2.3.18) org.glassfish.grizzly.websockets.ProtocolHandler has a bug when it close the webSocket. 

```
public GrizzlyFuture<DataFrame> close(int code, String reason) {
    return send(new ClosingFrame(code, reason),
            new EmptyCompletionHandler<DataFrame>() {

                @Override
                public void failed(final Throwable throwable) {
                    webSocket.onClose(null);
                }

                @Override
                public void completed(DataFrame result) {
                    if (!maskData) {
                        webSocket.onClose(null);
                    }
                }
            });
}
```

 The DataFrame is not set and when it comes to the GrizzlyAsyncHttpProvider it throws NullPointerException. 

```
public void onClose(org.glassfish.grizzly.websockets.WebSocket gWebSocket, DataFrame dataFrame) {
            try {
                if (ahcListener instanceof WebSocketCloseCodeReasonListener) {
                    ClosingFrame cf = ClosingFrame.class.cast(dataFrame);
                    WebSocketCloseCodeReasonListener.class.cast(ahcListener).onClose(webSocket, cf.getCode(), cf.getReason());
                } else {
                    ahcListener.onClose(webSocket);
                }
            } catch (Throwable e) {
                ahcListener.onError(e);
            }
}
```

**Comment by slandelle** [comment]
@oleksiys Can this be closed?

**Comment by laszlohordos** [comment]
I checked with the fixed grizzly-websockets-2.3.19 and it works well. In order to avoid this issue you must use at least async-http-client-1.8.16 #832 or async-http-client-1.9.12 #833 in other words the org.glassfish.grizzly:grizzly-http-client:1.8 is not compatible with grizzly-websockets-2.3.19 because that embeds async-http-client-1.8.11. 

I think this issue can be closed with this information.
